easycv/datasets/detection/data_sources/coco_livs.py

```diff
@@ -0,0 +1,143 @@
+            'http://images.cocodataset.org/zips/val2017.zip'
+        ],
+        train='lvis_v1_train.json',
+        val='lvis_v1_val.json',
+        dataset='images'
+        # default
+    )
+
+    def __init__(self,
+                 pipeline,
+                 path=None,
+                 download=True,
+                 split='train',
+                 test_mode=False,
+                 filter_empty_gt=False,
+                 classes=None,
+                 iscrowd=False,
+                 **kwargs):
+        """
+        Args:
+            path: This parameter is optional. If download is True and path is not provided,
+                    a temporary directory is automatically created for downloading
+            download: If the value is True, the file is automatically downloaded to the path directory.
+                      If False, automatic download is not supported and data in the path is used
+            split: train or val
+            test_mode (bool, optional): If set True, `self._filter_imgs` will not works.
+            filter_empty_gt (bool, optional): If set true, images without bounding
+                boxes of the dataset's classes will be filtered out. This option
+                only works when `test_mode=False`, i.e., we never filter images
+                during tests.
+            iscrowd: when traing setted as False, when val setted as True
+        """
+        if kwargs.get('cfg'):
+            self.cfg = kwargs.get('cfg')
+
+        assert split in ['train', 'val']
+        assert os.path.isdir(path), f'{path} is not dir'
+        self.lvis_path = Path(os.path.join(path, 'LVIS'))
+
+        if download:
+            self.download()
+
+        else:
+            if not (self.lvis_path.exists() and self.lvis_path.is_dir()):
+                raise FileNotFoundError(
+                    f'The data in the {self.lvis_path} file directory is not exists'
+                )
+
+        self.ann_file = str(self.lvis_path / self.cfg.get(split))
+        self.img_prefix = str(self.lvis_path / self.cfg.get('dataset'))
+        self.filter_empty_gt = filter_empty_gt
+        self.CLASSES = classes
+
+        # load annotations (and proposals)
+        self.data_infos = self.load_annotations(self.ann_file)
+        self.test_mode = test_mode
+        if not test_mode:
+            valid_inds = self._filter_imgs()
+            self.data_infos = [self.data_infos[i] for i in valid_inds]
+            self._set_group_flag()
+
+        self.iscrowd = iscrowd
+        self.max_labels_num = 120
+
+        transforms = []
+        for transform in pipeline:
+            if isinstance(transform, dict):
+                transform = build_from_cfg(transform, PIPELINES)
+                transforms.append(transform)
+            elif callable(transform):
+                transforms.append(transform)
+            else:
+                raise TypeError('transform must be callable or a dict')
+        self.pipeline = Compose(transforms)
```

Comment:
call DetSourceCoco.__init__

Reply:
no , coco is different from livs in init , some places are handled differently